src/coreclr/jit/assertionprop.cpp

```diff
@@ -1636,8 +1636,29 @@ AssertionIndex Compiler::optCreateAssertion(GenTree*         op1,
                 //
                 //  Copy Assertions
                 //
+                case GT_OBJ:
+                case GT_BLK:
+                {
+                    GenTree* const addr = op2->AsIndir()->Addr();
+
+                    if (addr->OperIs(GT_ADDR))
+                    {
+                        GenTree* const base = addr->AsOp()->gtOp1;
+
+                        if (base->OperIs(GT_LCL_VAR))
+                        {
+                            // layout compat?
+                            op2 = base;
+                            goto IS_COPY;
+                        }
+                    }
+
+                    goto DONE_ASSERTION;
+                }
+
```

Comment:
Is this still necessary after #71584?

(If it is, it should have a `TODO-ADDR: delete once <...>` comment)

Reply:
Will check once I pick up those changes.

Reply:
There are still some cases where it helps -- about 700 all told across SPMI
```
;;; String:Replace(ushort,ushort):String:this (MethodHash=5aa7fa78) #16249 in ASP.NET

fgMorphTree BB15, STMT00064 (before)
               [000326] -A---------                         *  ASG       simd32 (copy)
               [000324] D------N---                         +--*  LCL_VAR   simd32<System.Numerics.Vector`1[System.UInt16]> V29 tmp12        
               [000208] n----------                         \--*  OBJ       simd32<System.Numerics.Vector`1[System.UInt16]>
               [000207] -----------                            \--*  ADDR      byref 
               [000203] -------N---                               \--*  LCL_VAR   simd32<System.Numerics.Vector`1[System.UInt16]> V14 loc11

;; (with above)

GenTreeNode creates assertion:
               [000326] -A---------                         *  ASG       simd32 (copy)
In BB15 New Local Copy     Assertion: V29 == V14, index = #01

...
Assertion prop in BB15:
Copy     Assertion: V29 == V14, index = #01
               [000320] -----------                         *  LCL_VAR   simd32<System.Numerics.Vector`1[System.UInt16]> V14 loc11  

;; (without above)

no assertion, no copy prop        
```

Reply:
SIMD types, as one would suspect...

So the comment should be `TODO-ADDR: delete once local morph folds SIMD-typed indirections.` and we definitely need to check for layout compatibility.